### ISSUE TITLE
scm.subfolder support only one level of subfolders (not foo/baz/bar)

### ISSUE THREAD

**Issue by wizardsd** [conan-io/conan#3423]
conan v1.6.1, Windows 10, Python 3.6

I'm moving our codebase to conan packages and encountered following error. I have to specify not single directory in `scm.subfolder` attribute, but multiple ones:
```python
scm = {
        "type": "git",
        "subfolder": "foo/baz/bar",
        "url": "auto",
        "revision": "auto",
    }
``` 
The reason is that I shouldn't change paths in `#include` directives today and I will update paths in the future.

With such recipe conan fails to build missing packages, i.e.
`conan install . --build missing` fails with an strange error:
`FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\.conan2\\gk8g1phl\\1\\foo\\baz\bar`

I think that source of this problem is this line:
https://github.com/conan-io/conan/blob/develop/conans/client/tools/scm.py#L20

Could you call `os.makedirs` instead of `os.mkdir`?

- [*] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [*] I've specified the Conan version, operating system version and any tool that can be relevant.
- [*] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by memsharded** [comment]
We need to check it, but yes, it seems that changing to the ``os.makedirs()`` could fix this issue, lets have a look. Please upgrade to conan 1.7, it includes some updates to the SCM, so better check it with it too.

As a workaround, you might want to use 1 level subfolder, and then package the includes removing the unnecessary intermediate subfolders, something like:

```python
def package(self):
    self.copy("*.h", src="baz/bar", dst="include")
```

**Comment by wizardsd** [comment]
Thank you!
I've checked the issue and it exists in conan v1.7.0 too.

**Comment by wizardsd** [comment]
Yes, I'm copying .h files properly in `package` function. The problem is that header files improperly references themselves from sources in this library (like `#include <bla/foo/baz/bar/file.h` and I can't change it now) and other C++ modules references the same headers in the same way. So I have to set include paths as `self.cpp_info.includedirs = ["."]`

**Comment by memsharded** [comment]
Well, if you want to not have the headers in the root of the package you could always try something like:

```python
def package(self):
    self.copy("*.h", src="bla/foo/baz/bar", dst="include/bla/foo/baz/bar")
    #or maybe this works, because by default it keeps relative paths:
    self.copy("*.h", src="bla/", dst="include/bla/")
def package_info(self):
     self.cpp_info.includedirs = ["include"]
```

In any case, we will have a look to the subfolder scm issue, this is just an unrelated suggestion. Thanks!

**Comment by wizardsd** [comment]
Thank you, guys! It works as expected!
